Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XlgDebugger/JPath/JPath.g4

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// To generate lexer/parser/... run:
+//
+//   java -cp $CLASSPATH:antlr-4.7.2-complete.jar -Xmx500M \
+//      org.antlr.v4.Tool \
+//      -listener -visitor -Dlanguage=CSharp -package BuildXL.Execution.Analyzer.JPath JPath.g4
+
+grammar JPath;
+
+WS      : [ \t\r\n]+ -> skip    ; // skip spaces, tabs, newlines
+
+// logic operators
+NOT     : 'not';
+AND     : 'and';
+OR      : 'or' ;
+XOR     : 'xor';
+IFF     : 'iff';
+
+// bool operators
+GTE     : '>=' ;
+LTE     : '<=' ;
+GT      : '>'  ;
+LT      : '<'  ;
+EQ      : '='  ;
+NEQ     : '!=' ;
+MATCH   : '~'  ;
+NMATCH  : '!~' ;
+
+// arithmetic operators
+MINUS   : '-'  ;
+PLUS    : '+'  ;
+TIMES   : '*'  ;
+DIV     : '/'  ;
+MOD     : '%'  ;
+
+// array operators
+CONCAT    : '++' ;
+INTERSECT : '&'  ;
+
+IntLit
+    : [0-9]+ ;
+
+StrLit
+    : '\'' ~[']* '\''
+    | '"' ~["]* '"'
+    ;
+
+RegExLit
+    : '/' ~[/]+ '/' 
+    | '!' ~[!]+ '!'
+    ;
+
+fragment IdFragment
+    : [a-zA-Z][a-zA-Z0-9_]* ;
+
+PropertyId
+    : IdFragment ;
+
+VarId
+    : '$' IdFragment ;
+
+EscID
+    : '`' ~[`]+ '`' ;
+
+Opt
+    : '-' [a-zA-Z0-9'-']+ ;
+
+intBinaryOp
+    : Token=(PLUS | MINUS | TIMES | DIV | MOD) ;
+
+intUnaryOp
+    : Token=MINUS ;
+
+boolBinaryOp
+    : Token=(GTE | GT | LTE | LT | EQ | NEQ | MATCH | NMATCH) ;
+    
+logicBinaryOp
+    : Token=(AND | OR | XOR | IFF) ; 
+
+logicUnaryOp
+    : Token=NOT ;
+
+arrayBinaryOp
+    : Token=(CONCAT | INTERSECT) ;
+
+anyBinaryOp
+    : intBinaryOp
+    | boolBinaryOp
+    | logicBinaryOp
+    | arrayBinaryOp
+    ;
+
+intExpr
```

Comment:
surprised you haven't needed string expressions yet, how do you search for filenames?

Reply:
there are string literals and regex literals and there is the `~` operator which is a string/regex match.

But yes, in principle there should be either some string operators or the same can be done with string functions